addons/mail/static/src/js/composers/mention_manager.js

```diff
@@ -388,6 +388,8 @@ var MentionManager = Widget.extend({
             // mentions detection in text
             selectedSuggestion.name = selectedSuggestion.name.replace(/ /g, NON_BREAKING_SPACE);
             substitution = this._activeListener.delimiter + selectedSuggestion.name;
+            // suggestion after inserting is used over escaped content
+            selectedSuggestion.name = _.escape(selectedSuggestion.name);
```

Comment:
Ok thanks, updated. I changed the condition to `if (selectedSuggestion.name)` since anyway this would not be an issue for canned response and there is no `name` field on canned response.